resteasy-core/src/main/java/org/jboss/resteasy/plugins/delegates/NewCookieHeaderDelegate.java

```diff
@@ -35,12 +37,16 @@ public NewCookie fromString(String newCookie) throws IllegalArgumentException {
       Date expiry = null;
 
       ParameterParser parser = new ParameterParser();
-      Map<String, String> map = parser.parse(newCookie, ';');
+      LinkedHashMap<String, String> map = parser.parse(newCookie, ';');

```

Comment:
There is not need for this to be a `LinkedHashMap`. This line can remain unchanged.